### PR TITLE
gomod: update all

### PR DESCRIPTION
**Pull request by keegancsmith**
Also included a script so others can do this.

### CODE REVIEW

File: dev/go-mod-update.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -euxf -o pipefail
+
+cd "$(dirname "${BASH_SOURCE[0]}")/.." # cd to repo root dir
+
+export GOPRIVATE=github.com/mattn/go-sqlite3
```
Comment (unknwon) [inline dev/go-mod-update.sh]: Just curious, what does this line do and why do we need it?

**Comment by unknwon** [comment]
I was wondering... isn't the best time to update dependencies is the moment after branch cut? Thus we get a full iteration to test them during development.

**Comment by mrnugget** [comment]
> I was wondering... isn't the best time to update dependencies is the moment after branch cut? Thus we get a full iteration to test them during development.

Agree with that!

**Comment by keegancsmith** [comment]
Problem with "big" changes after branch cut is its effect on cherry-picking. Additionally in practice we seem to do the most vetting between branch cut and release. But I don't hold those opinions strongly, so I can hold off until after branch cut :)

**Comment by keegancsmith** [comment]
3.12 has been cut, ping.

**Pull request merged**